File: py/aoc2022/day1.py
"""
Day 1: Calorie Counting
"""

SAMPLE_INPUT = [
    "1000",
    "2000",
    "3000",
    "",
    "4000",
    "",
    "5000",
    "6000",
    "",
    "7000",
    "8000",
    "9000",
    "",
    "10000",
]


def _parse(lines):
    sums = []
    acc = 0
    for line in lines:
        line = line.strip()
        if line:
            acc += int(line)
        else:
            sums.append(acc)
            acc = 0
    sums.append(acc)
    return sums


def part1(lines):
    """
    >>> part1(SAMPLE_INPUT)
    24000
    """
    return max(_parse(lines))


def part2(lines):
    """
    >>> part2(SAMPLE_INPUT)
    45000
    """
    return sum(sorted(_parse(lines))[-3:])


parts = (part1, part2)

File: kt/src/commonMain/kotlin/com/github/ephemient/aoc2022/Day1.kt
package com.github.ephemient.aoc2022

class Day1(lines: List<String>) {
    private val sums = buildList {
        var sum = 0
        for (line in lines) {
            val value = line.toIntOrNull()
            if (value != null) {
                sum += value
            } else {
                add(sum)
                sum = 0
            }
        }
        add(sum)
    }

    fun part1(): Int = sums.max()

    fun part2(): Int = sums.sorted().takeLast(3).sum()
}
